src/debug.c

```diff
@@ -639,22 +644,25 @@ void debugCommand(client *c) {
             used = snprintf(nextra, remaining, " ql_compressed:%d", compressed);
             nextra += used;
             remaining -= used;
-            /* Add total uncompressed size */
-            unsigned long sz = 0;
-            for (quicklistNode *node = ql->head; node; node = node->next) {
-                sz += node->sz;
+            if (!fast) {
+                /* Add total uncompressed size */
+                unsigned long sz = 0;
+                for (quicklistNode *node = ql->head; node; node = node->next) {
+                    sz += node->sz;
+                }
+                used = snprintf(nextra, remaining, " ql_uncompressed_size:%lu", sz);
+                nextra += used;
+                remaining -= used;
             }
-            used = snprintf(nextra, remaining, " ql_uncompressed_size:%lu", sz);
-            nextra += used;
-            remaining -= used;
         }
 
-        addReplyStatusFormat(c,
-                             "Value at:%p refcount:%d "
-                             "encoding:%s serializedlength:%zu "
-                             "lru:%d lru_seconds_idle:%llu%s",
-                             (void *)val, val->refcount, strenc, rdbSavedObjectLen(val, c->argv[2], c->db->id),
-                             val->lru, estimateObjectIdleTime(val) / 1000, extra);
+        sds s = sdsempty();
+        s = sdscatprintf(s, "Value at:%p refcount:%d encoding:%s", (void *)val, val->refcount, strenc);
+        if (!fast) s = sdscatprintf(s, " serializedlength:%zu", rdbSavedObjectLen(val, c->argv[2], c->db->id));
+        s = sdscatprintf(s, " lru:%d lru_seconds_idle:%llu", val->lru, estimateObjectIdleTime(val) / 1000);
+        s = sdscatprintf(s, "%s", extra);
+        addReplyStatusLength(c, s, sdslen(s));
+        sdsfree(s);
```

Comment:
Same logic, but avoid many small sdscatprintf and maybe smaller diff like this?

```suggestion
        sds s = sdsempty();
        if (!fast) s = sdscatprintf(s, "serializedlength:%zu ", rdbSavedObjectLen(val, c->argv[2], c->db->id));
        addReplyStatusFormat(c,
                             "Value at:%p refcount:%d "
                             "encoding:%s %s"
                             "lru:%d lru_seconds_idle:%llu%s",
                             (void *)val, val->refcount, strenc, s,
                             val->lru, estimateObjectIdleTime(val) / 1000, extra);
        sdsfree(s);
```

Reply:
well, your diff is smaller, mine seems to be more readable, i prefer mine, so in the future we can just keep appending new fields line by line, like this LFU one #479 can be easy to add